### ISSUE TITLE
DFP_SERVICE_* ENV variables not applied?

### ISSUE THREAD

**Issue by narcoticfresh** [vfarcic/docker-flow-proxy#290]
## My need

I want to use the new `serviceDomainAlgo` setting (as introduced through #289). I want to set my default value on the container so it always uses that one.

## My compose file

```yml
version: '3.2'

services:

  proxy:
    image: vfarcic/docker-flow-proxy:17.07.13-2
    ports:
      - 80:80
      - 88:88
    networks:
      - proxy
    environment:
      - LISTENER_ADDRESS=swarm-listener
      - DEFAULT_PORTS=80
      - MODE=swarm
      - DEBUG=true
      - DFP_SERVICE_SERVICE_DOMAIN_ALGO=req.ssl_sni
    deploy:
      placement:
        constraints: [node.role == manager]

  swarm-listener:
    image: vfarcic/docker-flow-swarm-listener
    networks:
      - proxy
    volumes:
      - type: bind
        source: /var/run/docker.sock
        target: /var/run/docker.sock
    environment:
      - DF_NOTIFY_CREATE_SERVICE_URL=http://proxy:8080/v1/docker-flow-proxy/reconfigure
      - DF_NOTIFY_REMOVE_SERVICE_URL=http://proxy:8080/v1/docker-flow-proxy/remove
    deploy:
      placement:
        constraints: [node.role == manager]

  nginx:
    image: nginx:alpine
    networks:
      - proxy
    deploy:
      labels:
        - "com.df.notify=true"
        - "com.df.distribute=true"
        - "com.df.serviceDomain=test.local"
        - "com.df.port=80"
        - "com.df.srcPort=88"
        - "com.df.reqMode=tcp"
      placement:
        constraints:
          - node.role == manager

networks:
  proxy:
      driver: overlay
  default:
```

Please note the `DFP_SERVICE_SERVICE_DOMAIN_ALGO` env on the proxy service.

## The problem

After deploy, i check the ENV in the container:

```sh
user@host:~$ docker exec proxy_proxy.1.yth4e0eupbxr4vi4vx0dez0x4 ash -c export | grep DFP_
export DFP_SERVICE_SERVICE_DOMAIN_ALGO='req.ssl_sni
```
Very good.. let's see what the listener did:

```sh 
user@host:~$ docker logs proxy_swarm-listener.1.n0dzvt6n5llj6u4wnhkyp1do0
[..]
2017/07/14 08:15:04 Sending service created notification to http://proxy:8080/v1/docker-flow-proxy/reconfigure?distribute=true&port=80&reqMode=tcp&serviceDomain=test.local&serviceName=proxy_nginx&srcPort=88
```

Now let's check the frontend definition `/cfg/haproxy.cfg`:

```
frontend tcpFE_88                                       
    bind *:88                                           
    mode tcp                                                                                                                   
    option tcplog                                       
    log global                                          
    acl domain_proxy_nginx80 hdr(host) -i test.local                                                                             
    use_backend proxy_nginx-be80 if domain_proxy_nginx80
```

Hm, sadly still `hdr(host)` is used ;-/

## Assumption

I don't think this is a problem relating to `SERVICE_DOMAIN_ALGO` - yesterday i tried to set the frontend template file the same way - this also didn't work.

I've seen in the diff for the feature for #289 - the value is supposed to be read out of the ENV..

I thought it's a general problem, but strangely the `DEBUG` env is doing what it is supposed to do..

Any ideas?

**Comment by vfarcic** [comment]
The idea behind `DFP_SERVICE_*` env. vars. is to have an alternative method to service labels. If I'd take your service as an example, you could remove `com.df.*` labels from `nginx` and specify them as env. vars. of the `proxy`. They would be something like:

```
    environment:
      ...
      - DFP_SERVICE_1_SERVICE_DOMAIN=test.local
      - DFP_SERVICE_1_PORT=80
      - DFP_SERVICE_1_SRC_PORT=88
      - DFP_SERVICE_1_REQ_MODE=88
      - DFP_SERVICE_1_SERVICE_DOMAIN_ALGO=req.ssl_sni
```

That feature is very rarely used. It was added for some border-case.

My guess is that you want to specify `serviceDomainAlgo` on the `proxy` so that it is applied to all services. Can you please let me know if that's the case.

**Comment by narcoticfresh** [comment]
@vfarcic 

> My guess is that you want to specify serviceDomainAlgo on the proxy so that it is applied to all services. Can you please let me know if that's the case.

Yes, that's exactly the case.. 

An alternative would be a change to `swarm-listener` to attach arbitrary params to its HTTP request, but I couldn't find that there. If I could set i.e. `DF_ADDED_RECONFIGURE_PARAMS` to the `swarm-listener` container - that ENV would contain just `&param=1&param2=2` and would just be appended and sent..

Whatever would be easier..

**Comment by vfarcic** [comment]
It's available in the [release 17.07.14-5](https://github.com/vfarcic/docker-flow-proxy/releases/tag/17.07.14-5). Can you please try it out and let me know if it works as expected?
